### PR TITLE
bzip2: Conan V2 improvements

### PR DESCRIPTION
**Pull request by jwillikers**
Use export patches method.
Safely delete fPIC option.
Use Python F-Strings.

Specify library name and version:  **bzip2/***

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by prince-chrismc — COMMENTED** [review]
Test package needs can run :)

Bunch of linter comments

**Comment by jwillikers** [comment]
> Test package needs can run :)
> 
> Bunch of linter comments

Nice catch, thanks! Done.

**Review by prince-chrismc — APPROVED** [review]
🚀

**Review by SpaceIm — APPROVED** [review]

File: recipes/bzip2/all/CMakeLists.txt [hunk 1]
```diff
@@ -1,44 +1,34 @@
...
+install(TARGETS bz2)
 
 if(BZ2_BUILD_EXE)
-    add_executable(${CMAKE_PROJECT_NAME} ${SOURCE_SUBFOLDER}/bzip2.c)
-    target_link_libraries(${CMAKE_PROJECT_NAME} ${BZ2_LIBRARY})
-    target_include_directories(${CMAKE_PROJECT_NAME} PRIVATE ${SOURCE_SUBFOLDER})
-    list(APPEND BZ2_TARGETS ${CMAKE_PROJECT_NAME})
+    add_executable(bzip2 ${BZ2_SRC_DIR}/bzip2.c)
+    target_link_libraries(bzip2 PRIVATE bz2)
+    install(TARGETS bzip2)
```
Comment (SpaceIm) [inline recipes/bzip2/all/CMakeLists.txt:33]: don't use this form of install, it's broken for iOS, tvOS and watchOS

Comment (jwillikers) [reply recipes/bzip2/all/CMakeLists.txt]: Got it. Done.

File: recipes/bzip2/all/conanfile.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
-from conan import ConanFile
+from conan import ConanFile, conan_version
```
Comment (prince-chrismc) [inline recipes/bzip2/all/conanfile.py:1]: This should not be used

```suggestion
from conan import ConanFile
```

**Review by prince-chrismc — CHANGES_REQUESTED** [review]

File: recipes/bzip2/all/conanfile.py (hunk 2 above)
Comment (jwillikers) [reply recipes/bzip2/all/conanfile.py]: You got it!

File: recipes/bzip2/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -2,12 +2,22 @@ cmake_minimum_required(VERSION 3.1)
...
-message("BZIP2_NEED_PREFIX: ${BZIP2_NEED_PREFIX}")
-message("BZIP2_INCLUDE_DIRS: ${BZIP2_INCLUDE_DIRS}")
-message("BZIP2_INCLUDE_DIR: ${BZIP2_INCLUDE_DIR}")
-message("BZIP2_LIBRARIES: ${BZIP2_LIBRARIES}")
-message("BZIP2_VERSION_STRING: ${BZIP2_VERSION_STRING}")
 
 add_executable(test_package test_package.c)
 target_link_libraries(test_package PRIVATE BZip2::BZip2)
+
+set(_custom_vars
```
Comment (SpaceIm) [inline recipes/bzip2/all/test_package/CMakeLists.txt:9]:
```suggestion
# Test whether variables from https://cmake.org/cmake/help/latest/module/FindBZip2.html are properly defined
set(_custom_vars
```

To track the reference on which tests are based

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**